### PR TITLE
Amazon

### PR DESCRIPTION
**Pull request by Claire56**
Hi Sergey
Is there a more efficient way of answering this question

### CODE REVIEW

File: CGF.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+#given a list find the greatest commmon factor,( the largest number that is divisible by all 
+# items in the list
+
+def cdf(arr):
+	max_num = max(arr)
+	for i in range(max_num +1,1,-1):
```
Comment (shevelevs) [inline CGF.py:7]: why is this max_num + 1?

File: CGF.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+		if all(j%i == 0 for j in arr):
+			return i
+
+print(cdf([9,3,15]))
```
Comment (shevelevs) [inline CGF.py:11]: When you include test cases, you need to provide not only the input, but the expected result as well.
Normally you would do something like
```assert cdf([9, 3, 15]) == <expected value>```

File: CGF.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+
+#given a list find the greatest commmon factor,( the largest number that is divisible by all
```
Comment (shevelevs) [inline CGF.py:2]: I think this problem statement doesn't make sense to me, where did you get it from? Greatest common factor is the _largest number that divides all items in the list_. In your problem statement it asks for "largest number that is divisible by all items in the list", which does not exist (i.e. it's the same as asking what's the largest number possible).

File: CGF.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+
+#given a list find the greatest commmon factor,( the largest number that is divisible by all 
+# items in the list
+
+def cdf(arr):
```
Comment (shevelevs) [inline CGF.py:5]: what is the return value of this function for [2, 3]?

**Review by shevelevs — CHANGES_REQUESTED** [review]
1) Problem statement doesn't really make sense to me
2) If the problem is to find greatest common divisor, then your function is not working for some edge cases (what are the edge cases anyway? you should think about testing more)
3) Tests should not only include inputs, but expected outputs
4) What's the runtime of your solution?

File: CGF.py (hunk 4 above)
Comment (shevelevs) [inline CGF.py:5]: why is this function called "cdf"? if it's calculating greatest common divisor, this abbreviation doesn't make sense to me... am I missing something?
